### PR TITLE
DDMAINT-19626 - Created a new string cfc to escape JSON characters an…

### PR DESCRIPTION
**Pull request by kishore12**
…d also added a unit test.

### CODE REVIEW

File: String.cfc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+component {
+	string function escapeJSONcharacters(required string data) {
```
Comment (timmixell) [inline String.cfc]: i'd propose something like ```string function encodeForJSON(required string string)```

File: tests/TestString.cfc [hunk 2]
```diff
@@ -4,10 +4,10 @@ component extends = "mxunit.framework.TestCase" {
 		variables.string = new util.String();
 	}
 
-	function testEscapeJSONcharacters() {
+	function testEncodeForJSON() {
```
Comment (timmixell) [inline tests/TestString.cfc]: i've been doing test_functionName to isolate the method name from the "test" prefix - i'm not sure what best practice is, but you'll see that treatment elsewhere. additionally - all the OSS code is assumed to live within a "lib" package - so your unit test should reflect that (on line 4 - lib.util.String) - if someone runs the full util suite, they'll have a bad time.

File: String.cfc [hunk 3]
```diff
@@ -1,5 +1,5 @@
 component {
-	string function escapeJSONcharacters(required string data) {
+	string function encodeForJSON(required string data) {
```
Comment (timmixell) [inline String.cfc]: i think ```data``` is a holdout from your initial pass. i think the argument should/could be ```required string string``` or ```required string value``` (i prefer ```string``` but maybe let someone else chime in)

**Review by timmixell — COMMENTED** [review]
@kishore12 - a couple minor feedback items - i'd like you to nag someone else to review and comment before you bother making these changes <3

File: tests/TestString.cfc (hunk 2 above)
Comment (brisk02) [reply tests/TestString.cfc]: I second the test_functionName convention

**Review by wesme1337 — APPROVED** [review]

File: String.cfc [hunk 4]
```diff
@@ -1,5 +1,5 @@
 component {
-	string function encodeForJSON(required string data) {
+	string function encodeForJSON(required string string) {
 		// Remove ASCII characters 0-31
 		local.filteredData = reReplaceNoCase(arguments.data, "[\x00-\x1F]", " ", "all");
```
Comment (timmixell) [inline String.cfc]: @kishore12 - (notsureif) you ran your unit tests after makin' the changes...

**Review by brisk02 — APPROVED** [review]

**Review by timmixell — APPROVED** [review]

**Pull request merged**